awx_collection/plugins/module_utils/controller_api.py

```diff
@@ -531,7 +531,16 @@ def make_request(self, method, endpoint, *args, **kwargs):
                 controller_version = response.info().getheader('X-API-Product-Version', None)
 
             parsed_collection_version = Version(self._COLLECTION_VERSION).version
-            if not controller_version:
+            unauthenticated_endpoints = [
+                '/api/',
+                '/api',
+                '/api/v2/',
+                '/api/v2',
+                '/ping',
+                '/ping/',
+                '/authtoken/'
+            ]
```

Comment:
What I'm concerned about is that someone is unauthenticated (or in the process of authenticating), makes a request, logically gets `None` for `controller_version`, and thus drops into the `else:` here which first does `parsed_controller_version = Version(controller_version).version`. Importing the class used:

```python
>>> from ansible.module_utils.compat.version import LooseVersion as Version
>>> Version(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/alancoding/repos/ansible/lib/ansible/module_utils/compat/version.py", line 328, in __repr__
    return "LooseVersion ('%s')" % str(self)
                                   ^^^^^^^^^
  File "/home/alancoding/repos/ansible/lib/ansible/module_utils/compat/version.py", line 325, in __str__
    return self.vstring
           ^^^^^^^^^^^^
AttributeError: 'LooseVersion' object has no attribute 'vstring'
```

This would likely error and cause trouble.

As I read this, I highly suspect the original author treated a `None` value as the case where it was running against an _old_ version which obviously did not have the header. That is less of a concern now, as the header has been in released versions for years. The safest option may be to do nothing (no warning) in the None case.